lib/workload/stateful/database/component.ts

```diff
@@ -16,59 +18,120 @@ type MonitoringProps = {
   defaultDatabaseName: string;
+  /**
+   * Parameter group for the database.
+   */
   parameterGroupName: string;
+  /**
+   * Database username.
+   */
   username: string;
+  /**
+   * Database security group name.
+   */
+  securityGroupName: string;
+  /**
+   * Database secret name.
+   */
   masterSecretName: string;
+  /**
+   * Database engine version.
+   */
   version: rds.AuroraPostgresEngineVersion;
+  /**
+   * Number of database instances.
+   */
   numberOfInstance: number;
+  /**
+   * Min ACU for serverless database.
+   */
   minACU: number;
+  /**
+   * Max ACU for serverless database.
+   */
   maxACU: number;
+  /**
+   * Port to run the database on.
+   */
   dbPort: number;
+  /**
+   * The database removal policy.
+   */
   removalPolicy: RemovalPolicy;
-  allowedInboundSG?: ec2.SecurityGroup;
+  /**
+   * Create an inbound security group that can connect to the database. Stateless resources can recreate
+   * this security group to access the database.
+   */
+  inboundSecurityGroupName: string;
 };
 
-export class DatabaseConstruct extends Construct {
-  readonly dbSecurityGroup: ec2.SecurityGroup;
-  readonly dbCluster: rds.DatabaseCluster;
+/**
+ * Database props with a vpc.
+ */
+export type DatabaseProps = ConfigurableDatabaseProps & {
+  /**
+   * The database VPC.
+   */
+  vpc: ec2.IVpc;
+};
+
+export class Database extends Construct {
+  readonly securityGroup: SecurityGroup;
+  readonly inboundSecurityGroup: SecurityGroup;
+  readonly cluster: DatabaseCluster;
 
-  constructor(scope: Construct, id: string, vpc: ec2.IVpc, props: DatabaseProps) {
+  constructor(scope: Construct, id: string, props: DatabaseProps) {
     super(scope, id);
 
     const dbSecret = new rds.DatabaseSecret(this, id + 'DbSecret', {
       username: props.username,
       secretName: props.masterSecretName,
     });
 
-    this.dbSecurityGroup = new ec2.SecurityGroup(this, 'DbSecurityGroup', {
-      vpc: vpc,
+    this.securityGroup = new ec2.SecurityGroup(this, 'DbSecurityGroup', {
+      vpc: props.vpc,
       allowAllOutbound: false,
       allowAllIpv6Outbound: false,
+      securityGroupName: props.securityGroupName,
       description: 'security group for OrcaBus RDS',
     });
 
-    // give compute sg to access the rds
-    if (props.allowedInboundSG) {
-      this.dbSecurityGroup.addIngressRule(
-        props.allowedInboundSG,
-        ec2.Port.tcp(props.dbPort),
-        'allow the OrcaBus compute sg to access db'
-      );
-    }
+    this.inboundSecurityGroup = new ec2.SecurityGroup(this, 'DbInboundSecurityGroup', {
+      vpc: props.vpc,
+      allowAllOutbound: false,
+      allowAllIpv6Outbound: false,
+      securityGroupName: props.inboundSecurityGroupName,
+      description: 'an inbound security group to connect to the OrcaBus RDS',
+    });
+
+    this.securityGroup.addIngressRule(
+      this.inboundSecurityGroup,
+      ec2.Port.tcp(props.dbPort),
+      'allow the OrcaBus security group to access db'
+    );
```

Comment:
I'm not sure exactly what the best approach is with these security groups, but I figure there should be an "inbound" security group which the stateless can lookup.

Reply:
Ahh I see, that makes a lot of sense. Thanks! So just confirming, the idea behind the `SecurityGroupConstruct` is that stateless constructs will use this to connect to the database?

E.g. for filemanager, it will lookup `SecurityGroupConstruct` using the `securityGroupName`, and then pass this into the Lambda function's security groups [here](https://github.com/umccr/orcabus/blob/360b2b3efcd347e3297a04e2381a20b8cc06deee/lib/workload/stateless/filemanager/deploy/constructs/functions/function.ts#L116-L117)?

Reply:
Correct, yes.!